contrib/hadoop-store-builder/src/java/voldemort/store/readonly/mr/utils/VoldemortUtils.java

```diff
@@ -144,8 +144,8 @@ public static StoreDefinition getStoreDef(String xml) {
         return new StoreDefinitionsMapper().readStore(new StringReader(xml));
     }
 
-    public static String modifyURL(String originalUrl, String newProtocol, int newPort) {
-        if (newProtocol == null || newProtocol.isEmpty() || newPort < 0) {
+    public static String modifyURL(String originalUrl, String newProtocol, Integer newPort) {
+        if (newProtocol == null || newProtocol.isEmpty() || newPort == null) {
```

Comment:
Do we still want to check if newPort<0 ?  I don't fully grasp the point of your change, but there might be a regression if some code is still expecting to get away with setting a port as -1.